pallets/file-system/src/tests.rs

```diff
@@ -6690,6 +6690,287 @@ mod delete_file_and_pending_deletions_tests {
+                assert_eq!(
+                    file_system::PendingFileDeletionRequests::<Test>::get(owner_account_id.clone()),
+                    BoundedVec::<_, <Test as file_system::Config>::MaxUserPendingDeletionRequests>::try_from(
+                        vec![(file_key, bucket_id)]
+                    )
+                        .unwrap()
+                );
+
+                let pending_file_deletion_request_ttl: u32 =
+                    PendingFileDeletionRequestTtl::<Test>::get();
+                let pending_file_deletion_request_ttl: BlockNumberFor<Test> =
+                    pending_file_deletion_request_ttl.into();
+                let expiration_block = pending_file_deletion_request_ttl + System::block_number();
+
+                // Assert that the pending file deletion request was added to storage
+                assert_eq!(
+                    file_system::FileDeletionRequestExpirations::<Test>::get(expiration_block),
+                    vec![FileDeletionRequestExpirationItem {
+                        user: owner_account_id.clone(),
+                        file_key,
+                        bucket_id,
+                        file_size: size,
+                    }]
+                );
+
+                // Delete the bucket manually
+                pallet_storage_providers::Buckets::<Test>::remove(bucket_id);
+
+                // Roll past the expiration block
+                roll_to(pending_file_deletion_request_ttl + 1);
+
+                // Item expiration should be removed
+                assert_eq!(
+                    file_system::FileDeletionRequestExpirations::<Test>::get(expiration_block),
+                    vec![]
+                );
+
+                // Asser that the pending file deletion request was removed from storage
+                assert_eq!(
+                    file_system::PendingFileDeletionRequests::<Test>::get(owner_account_id.clone()),
+                    BoundedVec::<_, <Test as file_system::Config>::MaxUserPendingDeletionRequests>::default()
+                );
+
+                // Assert that the bucket size was not decreased because the bucket was not found
+                System::assert_has_event(Event::FailedToDecreaseBucketSize {
+                    user: owner_account_id.clone(),
+                    bucket_id,
+                    file_key,
+                    file_size: size,
+                    error: pallet_storage_providers::Error::<Test>::BucketNotFound.into(),
+                }.into());
+
+                // Assert that there was no queued priority challenge for file key in proofs dealer pallet
+                assert!(!pallet_proofs_dealer::PriorityChallengesQueue::<Test>::get()
+                .iter()
+                .any(|x| *x == (file_key, Some(TrieRemoveMutation))),);
+            });
+        }
+
+        #[test]
+        fn delete_file_expired_pending_file_deletion_request_with_insolvent_user() {
```

Comment:
Good that you added this test. It will be helpful then to check if we change the behaviour of this.